### PR TITLE
Axom: Remove blueos check on cuda variant

### PR DESCRIPTION
**Pull request by white238**
Removes an unnecessary check against powerpc/blueos for cuda variant on Axom.  This allowed us to build axom+cuda on a personal laptop.

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
Waiting for @becker33 approval (I just noticed there was a request for his review).

**Review by becker33 — DISMISSED** [review]

**Comment by alalazo** [comment]
@white238 There's one axom build failing in the Gitlab pipeline.

**Comment by white238** [comment]
> 
> 
> @white238 There's one axom build failing in the Gitlab pipeline.

I only see the following error:

```
ERROR: Job failed (system failure): error dialing backend: dial tcp 192.168.101.4:10250: connect: connection refused
```

Am I missing something else?

**Comment by alalazo** [comment]
> Am I missing something else?

Not sure :thinking:  The pipeline has been retriggered, so let's see what happens.

**Comment by alalazo** [comment]
@white238 This seems to happen consistently. To debug, you can try to give this command:
```console
$ spack ci reproduce-build --working-dir <tmp/dir> https://gitlab.spack.io/spack/spack/-/jobs/428352/artifacts/download
[ ... ]
```
and follow the instructions. I'll try to do the same later if the issue is still unclear.

**Comment by white238** [comment]
I have tracked down why this broke.  Previous to this PR the cuda variant on axom was a noop unless you were on blueos.  The current test case that is failing never worked as intended and defaulted to `ENABLE_CUDA=OFF`.  @becker33 is going to push a fix to turn off this test case (i think).

**Comment by becker33** [comment]
fix (removing the axom+cuda spec from test suite) is up.

This PR is a first step towards making `axom+cuda target=x86_64:` work, but it doesn't cover every case. It does mean that `+cuda` actually means something for `axom target=x86_64:`, which is a positive step, but that did mean the existing build now (correctly) fails.

@white238 I've left the offending line in the file, just commented out. If you remember to, just uncomment it when the `axom+cuda` build is stable enough for x86_64 to be worth testing, and then we can turn it back on to make sure we don't get a regression.

**Review by becker33 — DISMISSED** [review]

**Comment by white238** [comment]
This also broke when you don't have a fortran compiler defined in your compiler spec.  fix coming. sigh.

**Review by becker33 — APPROVED** [review]

**Pull request merged**
